Modules/_testlimitedcapi/sys.c

```diff
@@ -106,6 +107,48 @@ sys_getxoptions(PyObject *Py_UNUSED(module), PyObject *Py_UNUSED(ignored))
     return Py_XNewRef(result);
 }
 
+static PyObject *
+sys_audit(PyObject *Py_UNUSED(module), PyObject *args)
+{
+    const char *event;
+    const char *argFormat;
+    PyObject *arg1 = NULL, *arg2 = NULL, *arg3 = NULL;
+
+    if (!PyArg_ParseTuple(args, "ss|OOO", &event, &argFormat, &arg1, &arg2, &arg3)) {
+        return NULL;
+    }
+
+    int result;
+    if (arg1 == NULL) {
+        result = PySys_Audit(event, argFormat);
+    } else if (arg2 == NULL) {
+        result = PySys_Audit(event, argFormat, arg1);
+    } else if (arg3 == NULL) {
+        result = PySys_Audit(event, argFormat, arg1, arg2);
+    } else {
+        result = PySys_Audit(event, argFormat, arg1, arg2, arg3);
+    }
+
+    RETURN_INT(result);
+}
+
+static PyObject *
+sys_audittuple(PyObject *Py_UNUSED(module), PyObject *args)
+{
+    const char *event;
+    PyObject *tuple_args;
+
+    if (!PyArg_ParseTuple(args, "sO", &event, &tuple_args)) {
```

Comment:
Make the second argument optional and NULL by default. This will allow you to test with NULL.

Use `z#` for the first argument. This will allow you to test with non-UTF-8 bytestring and with NULL.